password.go

```diff
@@ -17,9 +20,49 @@ var PasswordQuestionTemplate = `
 {{- color "green+hb"}}? {{color "reset"}}
 {{- color "default+hb"}}{{ .Message }} {{color "reset"}}`
 
+type filteredReadlinePasswordOutput struct {
+	len int
+}
+
+// Write will be called by readline to echo the masked characters to the console.  It assumes
+// full control of the output with prompting, but since survey is printing the prompt readline
+// does not play nice here.  Write is called with clear-line statements which need to be
+// ignored then it is called with the full buffered output. So if you type "hi" this is the
+// sequence that will be sent to Write
+// 0: "\033[J\033[2K\r"
+// 1: " \b"
+// 2: "\033[J\033[2K\r"
+// 3: "*" (this is the masked "h" from "hi")
+// 4: "\033[J\033[2K\r"
+// 5: "**" (this is the masked "hi" from "hi")
+func (o *filteredReadlinePasswordOutput) Write(p []byte) (n int, err error) {
```

Comment:
I understand your concern, we could do that.   The main reason for this refactor for me is to be able to generate `help` messages in the prompt directly.  So my current plan is to have something like a `Help string` property for each Prompt which will allow me to provide hints to prompts like:
``` 
    prompt := &survey.Input{
        Message: "Enter birthday:",
        Help: "Birthday format must be in Day/Month/Year",
    }
```
Then the prompt would look something like this by default:
```
? Enter birthday: [? for help] 
```
then if user enters `?` we rewrite the prompt as:
```
Birthday format must be in Day/Month/Year
? Enter birthday: 
```
So if we make Password the odd-one-out it would also not be able to used to participate in the Help messages.  I would more confident about leaving it out if we had a strategy/plan to pull it back in but short of purging readline, not sure what else we could do.